Ultrasonic sensor implementatie

mees- opened the pull request:
Ik denk dat de code nu eindelijk moet werken. Met dank aan een snippet dat ik van @Badkuip heb gekregen heb ik nu (hopelijk) eindelijk een werkende versie kunnen schrijven. Deze versie is wel synchronous en blockt de CPU dus voor max ~50ms. Qua scheduler is dit niet een ramp aangezien de andere tasks daarna direct worden uitgevoerd.

Zouden jullie dit ook nog willen testen met het python programma?

Ik heb hem getest met wat mock code die elke keer automatisch naar de cu poort stuurt en ik krijg data terug (al is het in`cu` allemaal unicode dus keeg ik `(((()%&!` :P).

Ik weet niet zeker of ik alle communicatie code (in `serial` en `dataHandler`) goed heb geschreven dus zouden jullie dat kunnen checken?

Excuus voor de late voltooiing van de implementatie, ik heb enorm zitten vechten met die ultrasoon sensor om hem werkend te krijgen. Ik heb onderschat hoeveel werk het zou zijn.

fixes #9

mees- committed:
change the timer the scheduler uses to timer 0

Timer/counter1 can now be used for the ultrasone sesnsor.
The timing has changed slightly because timer/counter 1 is 16-bit and
timer/counter 0 is 8-bit so the prescaler has been changed to 1024

mees- committed:
add the files

I really hope this implementation will finally work

mees- committed:
Add some more code for using ultrasonic sensor

In this commit I added the buffers and helper methods for sending
ultrasonic data to the control centre.

mees- committed:
Fix error with updateSensorData adding distance

mees- committed:
Fix timing error with scheduler

mees- committed:
Rename function to reflect timer change in 2a61f17

mees- committed:
Add distance stuff to header files

mees- committed:
Add distance task

mees- committed:
Add distance to status reportin

mees- committed:
Fix main.h and other compiler errors

LutzenH committed:
Add serial-decoding for ultrasonic in Basestation

LutzenH commented:
Oke, het werkt, mooi zo, Er is nog wel een dingetje: [Line 1](https://github.com/LutzenH/ProjectEmbeddedSystems/blob/7130125869b4f47bba3d4ceb8efc78c052705cdd/Unit/src/rollerShutter.c#L86) en [Line 2](https://github.com/LutzenH/ProjectEmbeddedSystems/blob/7130125869b4f47bba3d4ceb8efc78c052705cdd/Unit/src/rollerShutter.c#L95)

Het updaten van de rollerShutter status i.v.m hoe ver het scherm dicht is moet aangegeven worden met de animatie, ThowV heeft het werk hier al voor gedaan er moet alleen nog een keuze worden gemaakt op basis van de uitrol-lengte van het scherm. Dit moet worden gedaan op basis van een minimale en maximale uitrollengte. waarbij bij de minimale waarde het scherm dichtgerold is en de maximale waarde uitgerold.

> De maximale uitrol van een scherm is instelbaar. Hetzelfde geldtvoor het oprollen. Bv het scherm kan tot 1.60m worden uitgerold en het oprollen moet stoppen als het uiteinde de ultrasonoorsensor tot 0.05mis genaderd.

Deze waardes moeten dynamisch gekozen kunnen worden, de knoppen zitten al in de basestation, De commando's moeten alleen nog geserialiseerd worden en vervolgens gewijzigd kunnen worden op de Module, voor dit gedeelte neem ik wel verantwoordelijkheid, als jij (of iemand anders) dan de minimale en maximale waardes kan verwerken en hier functies voor kan maken die ik kan aanroepen met seriele-commando's.

LutzenH committed:
Add Instruction Codes for minimal and maximum shutter-length

mees- commented:
> als jij (of iemand anders) dan de minimale en maximale waardes kan verwerken en hier functies voor kan maken die ik kan aanroepen met seriele-commando's.

Wat bedoel je hier precies mee?

mees- committed:
Add userPreferenceHandler functions

In order to store user preference in EEPROM with regards to shutter min
and max

mees- committed:
Add functions in serial so they're used correctly

mees- committed:
Change codes to reflect actions

mees- committed:
Implement using distance for rollsershutter

mees- commented:
Ik heb code toegevoegd die wat rollershutter dingen kan bepalen aan de hand van distance measurements in 3b5b1a3

mees- committed:
Add some small fixes

mees- committed:
Switch from overflow interrupt to 8 prescaler

LutzenH committed:
Add serial-command for Min and Max roll-out
